packages/ui/src/ui/components/Dialog/controls/OutputPathControl/ErasureCodecAttribute.tsx

```diff
@@ -0,0 +1,30 @@
+import React, {FC} from 'react';
+import {Attribute} from '../../../../store/reducers/navigation/modals/tableMergeSortModalSlice';
+import {SelectSingle} from '../../../Select/Select';
+import './ErasureCodecAttribute.scss';
+import cn from 'bem-cn-lite';
+
+const block = cn('erasure-codec-attribute');
+
+type Props = {
+    attribute: Attribute;
+    codecs: {value: string; text: string}[];
+    onChange: (attribute: Attribute) => void;
+};
+export const ErasureCodecAttribute: FC<Props> = ({attribute, codecs, onChange}) => {
+    if (!attribute.active) return null;
+
+    const handleChange = (value?: string) => {
+        onChange({
+            ...attribute,
+            value: value || '',
+        });
+    };
+
+    return (
+        <div className={block()}>
+            <div>Erasure codec:</div>
```

Comment:
Each individual component must have its own style settings in order to be autonomous. This makes it isolated and allows for reuse

Passing the class of the parent component makes sense for composition purposes. I mean margin. Since the component should not have them by default

I also don't like the idea of ​​using a grid parent. You are proposing to strip div components of their wrapper. But they can have more than 2 elements in their composition. This will cause problems later on. I don't agree with this approach

Reply:
> Each individual component must have its own style settings in order to be autonomous. This makes it isolated and allows for reuse

To be reusable all your components should have independent its own generally accepted set of props like `value`, `change(value) => void`. They are not individual components (`Optimized For Attribute`/`Compression Codec Attribute`/`ErasureCodecAttribute`) they are adaptors for `handleAttributeChange` from `OutputPathControl` and `attribute.active`. According to current set of props they will be never reused and the components looks redundant.

> I also don't like the idea of ​​using a grid parent. You are proposing to strip div components of their wrapper. But they can have more than 2 elements in their composition. This will cause problems later on. I don't agree with this approach

Current approach also doesn't avoid problems later, any new property with long label will break layout. And you will need to fix styles for your "individual" components.